Wurstpack/wurstscript/lib/systems/Knockback3.wurst

```diff
@@ -0,0 +1,240 @@
+        static var restitutionCoefficientGround       = .2
+        static var restitutionCoefficientDestructable = .3
+
+        // Ratio.
+        static var frictionCoefficientGround = .15
+
+        // In units per second squared.
+        static var gravity = 90.
+
+        // If a unit is not falling faster than this, it will not bounce. Units
+        // per second.
+        static var elasticityThreshold = -300.
+
+        // If a unit's vertical component is not greater than this, it will not
+        // become airborne. Units per second.
+        static var airborneThreshold = 150.
+
+        // For the purposes of friction, distinguish a unit which is airborne
+        // versus one which is sliding. Height in units.
+        static var isAirborneThreshold = 5.
+
+        // Below this speed, slighting units will stop.
+        static var minimumSlideSpeed = 30.
+
+        // Above this speed, sliding units will spawn a dust effect.
+        static var frictionFxThreshold = 180.
+        static var frictionFxPath = "Objects\\Spawnmodels\\Undead\\ImpaleTargetDust\\ImpaleTargetDust.mdl"
+
+        // Above this speed, destructables hit will be destroyed.
+        static var destroyDestructableSpeedThreshold = 300.
+
+        // Below this height, destructables hit may be destroyed.
+        static var destroyDestructableHeightThreshold = 150.
+
+        // The square rect size used for finding destructables.
+        @configurable static constant real DESTRUCTABLE_ENUM_SIZE = 130.
+
+        // If enabled, units have their move-speed changed while airborne.
+        // Warning: this is not a lock-safe form of crowd control.
+        @configurable static constant bool USE_MOVE_SPEED_MODIFIERS  = true
+
+        // If enabled, units have their prop window changed while airborne.
+        // Warning: this is not a lock-safe form of crowd control.
+        @configurable static constant bool USE_PROP_WINDOW_MODIFIERS = true
+
+
+        // Instance Variables
+        private unit u
+        private vec3 del
+
+
+        private static bool     hitDestructable
+        private static function destructableCallback()
+            let des = GetEnumDestructable()
+            if GetDestructableLife(des) > 0.
+                hitDestructable = true
+
+
+        private   static int            stackPointer     = -1
+        private   static thistype array knockStack
+        private   static timer          clock            = CreateTimer()
+        protected static rect           destructableRect = Rect(-1*Knockback3.DESTRUCTABLE_ENUM_SIZE, -1*Knockback3.DESTRUCTABLE_ENUM_SIZE, Knockback3.DESTRUCTABLE_ENUM_SIZE, Knockback3.DESTRUCTABLE_ENUM_SIZE)
+        protected static function       tick()
+            Knockback3 iter
+            var         index = 0
+            while index <= stackPointer
+                iter = knockStack[index]
+
+                let unitX     = iter.u.getX()
+                let unitY     = iter.u.getY()
```

Comment:
use u.getPos() and others to always obtain and calculate with vectors.
You shouldn't handle x/y reals at any point